zerver/lib/webhooks/common.py

```diff
@@ -1,13 +1,33 @@
+from django.conf import settings
 from django.http import HttpRequest
+from django.utils.translation import ugettext as _
 from typing import Optional, Text
 
 from zerver.lib.actions import check_send_stream_message, \
-    check_send_private_message
-from zerver.lib.exceptions import StreamDoesNotExistError
+    check_send_private_message, internal_send_private_message
+from zerver.lib.exceptions import StreamDoesNotExistError, JsonableError, \
+    ErrorCode
 from zerver.lib.request import REQ, has_request_variables
-from zerver.models import UserProfile
+from zerver.models import UserProfile, get_system_bot
 
 
+MISSING_EVENT_HEADER_MESSAGE = """
+Hi there! We thought you'd like to know that the integration associated
+with your bot {bot_name} just received a payload that is missing the
+HTTP header that allows us to determine the event type.
+"""
```

Comment:
This error message needs to be redone.  Think about the context of a user receiving this: You just setup your GitLab webhook and you get this error.  What are you supposed to do?

I think we want something like this:
```
Hi there!  Your bot {bot_name} just sent an HTTP request to {request.path} that is missing the 
HTTP {header_name} header.  Because this header is how {integration_name} indicates the event type,
this usually indicates a configuration issue, where you either entered the URL for a different integration.  
Contact {support_email} if you need help debugging!
```

Where `header_name` has the Django-specific `HTTP_` stripped from the front (possibly, rather than parsing, we should have the webhook code pass in `X_GITHUB_EVENT` and have the library code add the `HTTP_` on the end; I think that would simplify how this works for webhook authors).  

(Also, we should document this new function in the webhook walkthrough).